src/main/kotlin/no/nav/familie/ba/sak/kjerne/brev/BrevService.kt

```diff
@@ -302,23 +306,34 @@ class BrevService(
 
         val grunnlagOgSignaturData = hentGrunnlagOgSignaturData(vedtak)
 
-        val personopplysningGrunnlag = persongrunnlagService.hentAktivThrows(behandlingId = vedtak.behandling.id)
+        val behandlingId = vedtak.behandling.id
+        val personopplysningGrunnlag = persongrunnlagService.hentAktivThrows(behandlingId = behandlingId)
 
         val grunnlagForBegrunnelser = vedtaksperiodeService.hentGrunnlagForBegrunnelse(vedtak.behandling)
         val brevperioder =
-            vedtaksperioder.mapNotNull {
-                it.lagBrevPeriode(
-                    grunnlagForBegrunnelse = grunnlagForBegrunnelser,
-                    landkoder = integrasjonClient.hentLandkoderISO2(),
-                )
+            vedtaksperioder.mapNotNull { vedtaksperiode ->
+                try {
+                    vedtaksperiode.lagBrevPeriode(
+                        grunnlagForBegrunnelse = grunnlagForBegrunnelser,
+                        landkoder = integrasjonClient.hentLandkoderISO2(),
+                    )
+                } catch (e: BrevBegrunnelseFeil) {
+                    secureLogger.info(
+                        "Brevbegrunnelsefeil for behandling $behandlingId, " +
+                            "fagsak ${vedtak.behandling.fagsak.id} " +
+                            "på periode ${vedtaksperiode.fom} - ${vedtaksperiode.tom}. " +
+                            "\nAutogenerert test:\n" + testVerktøyService.hentBegrunnelsetest(behandlingId),
+                    )
+                    throw IllegalStateException(e.message, cause = e)
```

Comment:
```suggestion
                    throw IllegalStateException(e.message, e)
```

Kompilering feiler, det løser seg ved å fjerne bruk av named parameter for bare cause.

Reply:
Nice catch!